src/Sodaq_dataflash.h

```diff
@@ -26,6 +26,8 @@
 
 #define DF_AT45DB081D   1
 #define DF_AT45DB161D   2
+// Adding AT45DB041D:
+#define DF_AT45DB161D   3
 
 #define DF_VARIANT      DF_AT45DB161D
```

Comment:
Added the 3 lines.

I also tried to redefine the default AT45DB161D to AT45DB041D in a scetch:
.....
#include <Arduino.h>
#include <SPI.h>
#include <Sodaq_dataflash.h>
#define DF_VARIANT DF_AT45DB041D
......
I'm don't think the test had any value, but at least things work as expected.